p2p/src/protocol/peermgr.rs

```diff
@@ -260,21 +270,55 @@ impl<U: Handshake + SetTimeout + Connect + Disconnect + Events> PeerManager<U> {
 
     /// Initialize the peer manager. Must be called once.
     pub fn initialize<A: AddressSource>(&mut self, time: LocalTime, addrs: &mut A) {
-        let retry = self
+        let peers = self
             .config
-            .retry
+            .persistent
             .iter()
             .take(self.config.target_outbound_peers)
             .cloned()
             .collect::<Vec<_>>();
 
-        for addr in retry {
-            self.connect(&addr, time);
+        for addr in peers {
+            if !self.connect(&addr, time) {
+                panic!(
+                    "{}: unable to connect to persistent peer: {}",
+                    source!(),
+                    addr
+                );
+            }
         }
         self.upstream.set_timeout(IDLE_TIMEOUT);
         self.maintain_connections(addrs, time);
     }
 
+    fn retrier_add_peer(&mut self, addr: &net::SocketAddr, local_time: LocalTime) {
+        let attempts = self.retry_attempts.entry(*addr).or_default();
+        let delay = LocalDuration::from_secs(2_u64.pow((*attempts).min(63)))
```

Comment:
`2u64.saturating_pow(*attempts)` would be cleaner

Reply:
perfect